### PR TITLE
Test case Unhealthy VFs removed due to obsolescence in 4.4.

### PR DESCRIPTION
**Pull request by kononovn**
Added new test connectivity when PF in down.

### CODE REVIEW

File: test/conformance/tests/sriov_operator.go [hunk 1]
```diff
@@ -880,9 +880,9 @@ var _ = Describe("operator", func() {
 			})
 		})
 
-		Context("unhealthyVfs", func() {
-			// 25834
-			It(" Should not be able to create pod successfully if there are only unhealthy vfs", func() {
+		Context("PF shutdown", func() {
+			// 29398
+			It("Should be able to create pods successfully if PF is down.Pods are able to communicate with each other", func() {
```
Comment (SchSeba) [inline test/conformance/tests/sriov_operator.go:885]: nit: space before "Pods"

nit: "with each other" -> "with each other on the same node"

File: test/conformance/tests/sriov_operator.go [hunk 2]
```diff
@@ -899,7 +899,7 @@ var _ = Describe("operator", func() {
 				sriovDevice := unusedSriovDevices[0]
 
 				createSriovPolicy(sriovDevice.Name, testNode, 5, resourceName)
-				ipam := `{"type": "host-local","ranges": [[{"subnet": "3ffe:ffff:0:01ff::/64"}]],"dataDir": "/run/my-orchestrator/container-ipam-state"}`
+				ipam := `{"type":"host-local","subnet":"10.10.10.0/24","rangeStart":"10.10.10.171","rangeEnd":"10.10.10.181","routes":[{"dst":"0.0.0.0/0"}],"gateway":"10.10.10.1"}`
```
Comment (SchSeba) [inline test/conformance/tests/sriov_operator.go:902]: why you change the subnet to use ipv4?

File: test/conformance/tests/sriov_operator.go [hunk 3]
```diff
@@ -1079,28 +1099,7 @@ func createSriovPolicy(sriovDevice string, testNode string, numVfs int, resource
...
-	}, 3*time.Minute, 1*time.Second).Should(Equal(k8sv1.PodPending))
-	pod, err := clients.Pods(namespaces.Test).Get(createdPod.Name, metav1.GetOptions{})
-	Expect(err).ToNot(HaveOccurred())
-	for _, condition := range pod.Status.Conditions {
-		if condition.Reason == "Unschedulable" && strings.Contains(condition.Message, "Insufficient openshift.io/"+resourceName) {
-			return
-		}
-	}
-	Fail("Pod should be Unschedulable due to: Insufficient openshift.io/" + resourceName)
+	}, 15*time.Minute, time.Second).Should(Equal(int64(numVfs)))
```
Comment (SchSeba) [inline test/conformance/tests/sriov_operator.go:1102]: why you put a long wait here?

File: test/conformance/tests/sriov_operator.go (hunk 2 above)
Comment (kononovn) [reply test/conformance/tests/sriov_operator.go:902]: This test case doesn't related to ipv6. So i decided to use ipv4 protocol instead

File: test/conformance/tests/sriov_operator.go (hunk 3 above)
Comment (kononovn) [reply test/conformance/tests/sriov_operator.go:1102]: Sometimes 10 minutes is not enough for Node to update capacity.

File: test/conformance/tests/sriov_operator.go [hunk 4]
```diff
@@ -910,7 +910,34 @@ var _ = Describe("operator", func() {
...
+				Eventually(func() k8sv1.PodPhase {
+					secondPod, _ = clients.Pods(namespaces.Test).Get(secondPod.Name, metav1.GetOptions{})
+					return secondPod.Status.Phase
+				}, 3*time.Minute, 1*time.Second).Should(Equal(k8sv1.PodRunning))
+				secondPodIPs, err := network.GetSriovNicIPs(secondPod, "net1")
+				Expect(err).ToNot(HaveOccurred())
+				Expect(len(secondPodIPs)).To(Equal(1))
+
+				stdout, stderr, err := pod.ExecCommand(clients, firstPod, "ping", "-c", "4", secondPodIPs[0])
+				Expect(err).ToNot(HaveOccurred(), "Failed to ping second pod", stderr)
```
Comment (mmirecki) [inline test/conformance/tests/sriov_operator.go:939]: You can use "func pingPod(...)" to test for the ping.

**Comment by fedepaol** [comment]
@kononovn please rebase

File: test/conformance/tests/sriov_operator.go [hunk 5]
```diff
@@ -1450,11 +1465,15 @@ func createCustomTestPod(node string, networks []string, hostNetwork bool) *k8sv
...
+	ipProtocolVersion := "6"
+	if len(strings.Split(ip, ".")) == 4 {
+		ipProtocolVersion = "4"
+	}
 	podDefinition := pod.RedefineWithNodeSelector(
 		pod.RedefineWithRestartPolicy(
 			pod.RedefineWithCommand(
 				pod.DefineWithNetworks([]string{sriovNetworkAttachment}),
-				[]string{"sh", "-c", "ping -6 -c 3 " + ip}, []string{},
+				[]string{"sh", "-c", "ping -" + ipProtocolVersion + " -c 3 " + ip}, []string{},
```
Comment (SchSeba) [inline test/conformance/tests/sriov_operator.go:1476]: please use the `fmt.Sprint` here

File: test/conformance/tests/sriov_operator.go [hunk 6]
```diff
@@ -540,56 +540,15 @@ var _ = Describe("[sriov] operator", func() {
...
-				defer namespaces.DeleteAndWait(clients, ns2, 1*time.Minute)
-				err := namespaces.Create(ns1, clients)
-				Expect(err).ToNot(HaveOccurred())
-				err = namespaces.Create(ns2, clients)
-				Expect(err).ToNot(HaveOccurred())
-
-				sriovNetworkName := "test-sriovnetwork"
+		Context("PF shutdown", func() {
+			// 29398
+			It("Should be able to create pods successfully if PF is down.Pods are able to communicate with each other on the same node", func() {
```
Comment (SchSeba) [inline test/conformance/tests/sriov_operator.go:545]: I don't think this test is right

you need to search for an sriov interface that you can shutdown.

File: test/conformance/tests/sriov_operator.go [hunk 7]
```diff
@@ -1192,6 +1151,69 @@ var _ = Describe("[sriov] operator", func() {
...
+				Expect(err).ToNot(HaveOccurred())
+				Eventually(func() error {
+					netAttDef := &netattdefv1.NetworkAttachmentDefinition{}
+					return clients.Get(context.Background(), runtimeclient.ObjectKey{Name: "sriovnetwork", Namespace: namespaces.Test}, netAttDef)
+				}, 10*time.Second, 1*time.Second).ShouldNot(HaveOccurred())
+
+				macvlanNadName := "macvlan-nad"
+				nodeNicName := sriovInfos.States[testNode].Status.Interfaces[0].Name
+				macvlanNad := network.CreateMacvlanNetworkAttachmentDefinition(macvlanNadName, namespaces.Test, nodeNicName)
+				defer clients.Delete(context.Background(), &macvlanNad)
```
Comment (SchSeba) [inline test/conformance/tests/sriov_operator.go:1177]: you can move the defer after the creation

```
err = clients.Create(context.Background(), &macvlanNad)
Expect(err).ToNot(HaveOccurred())
```

File: test/conformance/tests/sriov_operator.go [hunk 8]
```diff
@@ -1192,6 +1151,69 @@ var _ = Describe("[sriov] operator", func() {
...
+				pod, err := clients.Pods(namespaces.Test).Create(firstPodDefinition)
+				Expect(err).ToNot(HaveOccurred())
+				Eventually(func() k8sv1.PodPhase {
+					pod, _ = clients.Pods(namespaces.Test).Get(pod.Name, metav1.GetOptions{})
+					return pod.Status.Phase
+				}, 3*time.Minute, 1*time.Second).Should(Equal(k8sv1.PodRunning))
+				podIPs, err := network.GetSriovNicIPs(pod, "net1")
+				Expect(err).ToNot(HaveOccurred())
+				Expect(len(podIPs)).To(Equal(1))
+				pingPod(podIPs[0], testNode, sriovNetworkName)
```
Comment (SchSeba) [inline test/conformance/tests/sriov_operator.go:1214]: please create this pod with two interfaces same as the first one and ping between both interfaces

**Pull request merged**